warehouse/api.go

```diff
@@ -87,27 +89,50 @@ type UploadAPIT struct {
 	dbHandle          *sql.DB
 	log               logger.LoggerI
 	connectionManager *controlplane.ConnectionManager
-	isHosted          bool
+	isMultiWorkspace  bool
 }
 
 var UploadAPI UploadAPIT
 
-func InitWarehouseAPI(dbHandle *sql.DB, log logger.LoggerI) {
-	workspaceToken := config.GetWorkspaceToken()
-	isMultiWorkspace := config.GetEnvAsBool("HOSTED_SERVICE", false)
-	if isMultiWorkspace {
-		workspaceToken = config.GetEnv("HOSTED_SERVICE_SECRET", "password")
+func InitWarehouseAPI(dbHandle *sql.DB, log logger.LoggerI) error {
+	deploymentType, err := deployment.GetFromEnv()
+	if err != nil {
+		pkgLogger.Errorf("error getting deployment type: %s", err.Error())
+		return err
+	}
+	var connectionIdentifier string
+	var isMultiWorkspace bool
+	switch deploymentType {
+	case deployment.DedicatedType:
+		connectionIdentifier = config.GetWorkspaceToken()
+	case deployment.MultiTenantType:
+		isMultiWorkspace = true
+		isNamespaced := config.IsEnvSet("WORKSPACE_NAMESPACE")
```

Comment:
Why are we reading environment variables from a completely different domain here in warehouse? Shouldn't we get what we need by accessing some kind of API (API as in go package)? WDYT?